cli/src/action/certs.rs

```diff
@@ -0,0 +1,389 @@
+        let server_key_path = private_cert_dir.join(SERVER_KEY);
+
+        // if check
+        if args.is_present("skip") {
+            if client_cert_path.exists()
+                && client_key_path.exists()
+                && server_cert_path.exists()
+                && server_key_path.exists()
+            {
+                println!("Certificates and keys exist");
+            } else if client_cert_path.exists()
+                || client_key_path.exists()
+                || server_cert_path.exists()
+                || server_key_path.exists()
+            {
+                // this will always be an error condition
+                return Err(CliError::ActionError(
+                    "Some but not all certificates and keys already exist".into(),
+                ));
+            } else {
+                println!("Certificates and keys do not exist");
+            }
+            return Ok(());
+        } else if client_cert_path.exists()
+            && client_key_path.exists()
+            && server_cert_path.exists()
+            && server_key_path.exists()
+            && !args.is_present("force")
+        {
+            return Err(CliError::ActionError(
+                "Certificates and keys already exist".into(),
+            ));
+        } else if (client_cert_path.exists()
+            || client_key_path.exists()
+            || server_cert_path.exists()
+            || server_key_path.exists())
+            && !args.is_present("force")
+        {
+            return Err(CliError::ActionError(
+                "Some but not all certificates and keys already exist".into(),
+            ));
+        }
+
+        let forcing = client_cert_path.exists()
+            && client_key_path.exists()
+            && server_cert_path.exists()
+            && server_key_path.exists()
+            && args.is_present("force");
+
+        create_cert(
+            cert_dir.to_path_buf(),
+            private_cert_dir,
+            common_name,
+            forcing,
+            args.is_present("quiet"),
+            true,
+        )?;
+
+        Ok(())
+    }
+}
+
+#[allow(clippy::too_many_arguments)]
+fn create_cert(
+    cert_path: PathBuf,
+    private_cert_path: PathBuf,
+    common_name: String,
+    forcing: bool,
+    quiet: bool,
+    change_permissions: bool,
+) -> Result<(), CliError> {
+    // Generate Certificate Authority keys and certificate.
+    // These files are not saved
+    let (ca_key, ca_cert) = make_ca_cert()?;
+    // Generate client and server keys and certificates
+
+    let (client_key, client_cert) = make_ca_signed_cert(&ca_cert, &ca_key, &common_name)?;
+
+    write_file(
+        cert_path.clone(),
+        CLIENT_CERT,
+        &client_cert.to_pem()?,
+        change_permissions,
+    )?;
+
+    write_file(
+        private_cert_path.clone(),
+        CLIENT_KEY,
+        &client_key.private_key_to_pem_pkcs8()?,
+        change_permissions,
+    )?;
+
+    if !quiet {
+        if forcing {
```

Comment:
You could refactor this a bit and put a "File will be overwritten (forced): {}" in the calling function and remove force boolean from this function.